File: tests/fixtures/java-project/src/main/java/com/example/AuthService.java
package com.example;

public class AuthService implements AuthProvider {
    @Override
    public User authenticate(String username, String password) {
        User user = findUser(username);
        if (user != null && verifyPassword(user, password)) {
            return user;
        }
        return null;
    }

    @Override
    public void revoke(Session session) {
        System.out.println("Session revoked: " + session.getToken());
    }

    public User login(String username, String password) {
        return authenticate(username, password);
    }

    public void logout(User user) {
        System.out.println("User logged out: " + user.getName());
    }

    private User findUser(String username) {
        return new User(username, username + "@example.com", Role.ADMIN);
    }

    protected boolean verifyPassword(User user, String password) {
        return password != null && !password.isEmpty();
    }
}

File: tests/fixtures/kotlin-project/src/main/kotlin/com/example/AuthService.kt
package com.example

class AuthService : AuthProvider {
    override fun authenticate(username: String, password: String): User? {
        val user = findUser(username)
        return if (user != null && verifyPassword(user, password)) user else null
    }

    override fun revoke(session: Session) {
        println("Session revoked: ${session.token}")
    }

    fun login(username: String, password: String): User? {
        return authenticate(username, password)
    }

    fun logout(user: User) {
        println("User ${user.name} logged out")
    }

    private fun findUser(username: String): User? {
        return User(username, "$username@example.com", Role.ADMIN)
    }

    internal fun verifyPassword(user: User, password: String): Boolean {
        return password.isNotEmpty()
    }

    companion object {
        fun create(): AuthService = AuthService()
    }
}
